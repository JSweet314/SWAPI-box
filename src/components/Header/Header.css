.header {
  width: 95%;
  margin: auto;
  border-bottom: 1px solid #AAAAAB;
}

.header__h1 {
  font-size: 3rem;
  text-align: center;
}

.header__favBtn {
  position: absolute;
  right: 2rem;
  top: 2rem;
  padding: 1rem;
  cursor: pointer;
}

.header__favCount {
  margin: 0 0 0 1rem;
  padding: 0.5rem;
  background-color: #4B545F;
  color: #fff;
}