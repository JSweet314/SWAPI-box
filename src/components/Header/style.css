.header {
  width: 100%;
  margin: auto;
  border-bottom: 1px solid #AAAAAB;
  background-color: #4B545F;
  position: relative;
  padding: 
}

.header__h1 {
  font-size: 5rem;
  color: orange;
  text-align: center;
  margin: 0;
  padding: 2rem 2rem 0 2rem;
}

.header__favBtn {
  background-color: gray;
  position: absolute;
  right: 2rem;
  top: 2rem;
  padding: 1rem;
  cursor: pointer;
  text-decoration: none;
  font-size: 1.5rem;
  color: white;
  border: 1px solid black;
}

.header__favBtn.active {
  background:#18212A;
  color: white;  
}

.header__favCount {
  margin: 0 0 0 1rem;
  padding: 0.5rem;
  background-color: #4B545F;
  color: #fff;
  font-size: 1.5rem;
}