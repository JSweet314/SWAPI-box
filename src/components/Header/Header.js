import React from 'react';
import NavLinks from '../NavLinks/index';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({ numberOfFavorites }) => {
  return (
    <header className="header">
      <h1 className="header__h1">SWAPI-Box</h1>
      <button className="header__favBtn">
        View Favorites 
        <span className="header__favCount">{numberOfFavorites}</span>
      </button>
      <NavLinks />
    </header>
  );
};

Header.propTypes = {
  numberOfFavorites: PropTypes.number.isRequired
};

export default Header;