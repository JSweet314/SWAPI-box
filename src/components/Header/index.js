import React from 'react';
import {NavLink} from 'react-router-dom';
import PropTypes from 'prop-types';
import './style.css';

const Header = ({numberOfFavorites}) => {
  return (
    <header className="header">
      <h1 className="header__h1">SWAPI-Box</h1>
      <ul className="nav">
        <li>
          <NavLink exact to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="/category/people">People</NavLink>
        </li>
        <li>
          <NavLink to="/category/planets">Planets</NavLink>
        </li>
        <li>
          <NavLink to="/category/vehicles">Vehicles</NavLink>
        </li>
      </ul>
      <NavLink className="header__favBtn" exact to="/category/favorites">
        View Favorites
        <span className="header__favCount">{numberOfFavorites}</span>
      </NavLink>
    </header>
  );
};

Header.propTypes = {
  numberOfFavorites: PropTypes.number.isRequired
};

export default Header;