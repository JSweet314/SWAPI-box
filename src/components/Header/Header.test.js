import React from 'react';
import Header from './Header';
import {shallow} from 'enzyme';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header numberOfFavorites={0} />);
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});