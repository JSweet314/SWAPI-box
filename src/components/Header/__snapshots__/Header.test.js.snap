// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        SWAPI-Box
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "SWAPI-Box",
      },
      "ref": null,
      "rendered": "SWAPI-Box",
      "type": "h1",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          SWAPI-Box
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
